assets/js/state/sagas/settings.test.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { recordSaga } from '@lib/test-utils';
+
+import { networkClient } from '@lib/network';
+import MockAdapter from 'axios-mock-adapter';
+
+import { apiKeySettingsFactory } from '@lib/test-utils/factories/settings';
+import { addDays, formatISO, subDays } from 'date-fns';
+import HealthIcon from '@common/HealthIcon';
+import { notify, dismissableNotify } from '@state/notifications';
+import { checkApiKeyExpiration } from './settings';
+
+const axiosMock = new MockAdapter(networkClient);
+
+describe('Settings sagas', () => {
+  beforeEach(() => {
+    axiosMock.reset();
+  });
+
+  describe('checkApiKeyExpiration saga', () => {
+    it('should skip any expiration toast creation when the expire_at of the api key is not defined', async () => {
+      axiosMock
+        .onGet('/api/v1/settings/api_key')
+        .reply(200, apiKeySettingsFactory.build({ expire_at: null }));
+
+      const dispatched = await recordSaga(checkApiKeyExpiration, {});
+
+      expect(dispatched).toEqual([]);
+    });
+
+    it('should dispatch an expired key notification if the expire_at of the api key is in the past', async () => {
```

Comment:
for me these test description texts are a bit awkard, but well, i can live with that XD